### PR TITLE
fix(cross-publish): added deprecation warnings and new flags support

### PR DESCRIPTION
**Pull request by vkalta**
This fixes CS-28252.

### CODE REVIEW

File: packages/contentstack-bulk-publish/src/commands/cm/bulk-publish/cross-publish.js [hunk 1]
```diff
@@ -125,18 +157,76 @@ CrossPublishCommand.flags = {
...
+  }),
+  'source-env': flags.string({
+    description: 'Source Env',
+  }),
+  destEnv: flags.string({
+    char: 'd',
+    description: 'Destination Environments',
+    multiple: true,
+    hidden: true,
+    parse: printFlagDeprecation(['--destEnv', '-e'], ['--environment']),
```
Comment (netrajpatel) [inline packages/contentstack-bulk-publish/src/commands/cm/bulk-publish/cross-publish.js:215]: -e is not configured with destEnv please correct it

**Review by netrajpatel — CHANGES_REQUESTED** [review]

**Comment by netrajpatel** [comment]
@vkalta Also update examples(CrossPublishCommand.examples) with valid long flags only

File: packages/contentstack-bulk-publish/src/commands/cm/bulk-publish/cross-publish.js (hunk 1 above)
Comment (vkalta) [reply packages/contentstack-bulk-publish/src/commands/cm/bulk-publish/cross-publish.js]: @netrajpatel I have made the changes. Please check.

File: packages/contentstack-bulk-publish/src/commands/cm/bulk-publish/cross-publish.js [hunk 2]
```diff
@@ -150,19 +237,19 @@ CrossPublishCommand.flags = {
...
   'csdx cm:bulk-publish:cross-publish --config [PATH TO CONFIG FILE]',
   'csdx cm:bulk-publish:cross-publish -c [PATH TO CONFIG FILE]',
   '',
-  'Using --retryFailed or -r flag',
-  'csdx cm:bulk-publish:cross-publish --retryFailed [LOG FILE NAME]',
+  'Using --retry-failed flag',
+  'csdx cm:bulk-publish:cross-publish --retry-failed [LOG FILE NAME]',
   'csdx cm:bulk-publish:cross-publish -r [LOG FILE NAME]',
   '',
   'Using --branch or -B flag',
```
Comment (netrajpatel) [inline packages/contentstack-bulk-publish/src/commands/cm/bulk-publish/cross-publish.js:251]: -B is not a valid short flag and to be removed

File: packages/contentstack-bulk-publish/src/commands/cm/bulk-publish/cross-publish.js [hunk 3]
```diff
@@ -150,19 +237,19 @@ CrossPublishCommand.flags = {
...
-  'csdx cm:bulk-publish:cross-publish -t [CONTENT TYPE] -e [SOURCE ENV] -d [DESTINATION ENVIRONMENT] -l [LOCALE] -a [MANAGEMENT TOKEN ALIAS] -x [DELIVERY TOKEN] -B [BRANCH NAME]',
+  'csdx cm:bulk-publish:cross-publish --content-type [CONTENT TYPE] --source-env [SOURCE ENV] --environments [DESTINATION ENVIRONMENT] --locales [LOCALE] -a [MANAGEMENT TOKEN ALIAS] --delivery-token [DELIVERY TOKEN] -B [BRANCH NAME]',
```
Comment (netrajpatel) [inline packages/contentstack-bulk-publish/src/commands/cm/bulk-publish/cross-publish.js:252]: Remove this -B as well

Comment (netrajpatel) [reply packages/contentstack-bulk-publish/src/commands/cm/bulk-publish/cross-publish.js:252]: and replace the valid branch flag

File: packages/contentstack-bulk-publish/src/commands/cm/bulk-publish/cross-publish.js [hunk 4]
```diff
@@ -125,18 +157,73 @@ CrossPublishCommand.flags = {
...
+    description: 'Delivery token for source environment',
+    hidden: true,
+    parse: printFlagDeprecation(['--deliveryToken', '-x'], ['--delivery-token']),
+  }),
+  'delivery-token': flags.string({
+    description: 'Delivery token for source environment',
+  }),
   config: flags.string({ char: 'c', description: 'Path to the config file' }),
   yes: flags.boolean({ char: 'y', description: 'Agree to process the command with the current configuration' }),
   branch: flags.string({
```
Comment (netrajpatel) [inline packages/contentstack-bulk-publish/src/commands/cm/bulk-publish/cross-publish.js:229]: Add deprecation warnings for branch short flag -B

File: packages/contentstack-bulk-publish/src/commands/cm/bulk-publish/cross-publish.js (hunk 3 above)
Comment (netrajpatel) [reply packages/contentstack-bulk-publish/src/commands/cm/bulk-publish/cross-publish.js:252]: @vkalta

**Review by netrajpatel — CHANGES_REQUESTED** [review]

**Review by netrajpatel — APPROVED** [review]

**Pull request merged**